src/tensor.h

```diff
@@ -0,0 +1,13 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+namespace Generators {
+
+struct Tensor : std::enable_shared_from_this<Tensor> {
+  Tensor() = default;
+  Tensor(std::unique_ptr<OrtValue> ort_tensor) : ort_tensor_{std::move(ort_tensor)} {}
```

Comment:
>std::unique_ptr<OrtValue> [](http://example.com/codeflow?start=9&length=25)

use std::unique_ptr<OrtValue> && to specify the ort_tensor object will be moved

Reply:
Amusingly it's recommended to pass by value to enforce that it's taking ownership, as it can only be passed by moving it into the parameter you pass. Vs rvalue reference "might" take ownership.

https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#r32-take-a-unique_ptrwidget-parameter-to-express-that-a-function-assumes-ownership-of-a-widget

R.32: Take a unique_ptr<widget> parameter to express that a function assumes ownership of a widget

Reason Using unique_ptr in this way both documents and enforces the function call’s ownership transfer.

Reply:
I prefer to use && to keep consistent.